layer/src/main/scala/geotrellis/layer/stitch/StitchCollectionMethods.scala

```diff
@@ -38,6 +37,52 @@ abstract class SpatialTileLayoutCollectionStitchMethods[
     val (ulx, uly) = (base.x - offsx.toDouble * layout.cellwidth, base.y + offsy * layout.cellheight)
     Raster(tile, Extent(ulx, uly - tile.rows * layout.cellheight, ulx + tile.cols * layout.cellwidth, uly))
   }
+
+  /**
+    * Stitch all tiles in the collection using a sparse stitch that handles missing keys
+    *
+    * Any missing tiles within the extent are filled with an empty prototype tile.
+    *
+    * @param extent The requested [[geotrellis.vector.Extent]] of the output [[geotrellis.raster.Raster]]
+    * @return The stitched Raster, otherwise None if the collection is empty or the extent does not intersect
+    */
+  def sparseStitch(extent: Extent): Option[Raster[V]] = {
+    if (self.headOption.isEmpty) {
+      None
+    } else {
+      val tile = self.head._2
+      val layoutDefinition = self.metadata.getComponent[LayoutDefinition]
+      val mapTransform = layoutDefinition.mapTransform
+      val expectedKeys = mapTransform(extent)
+        .coordsIter
+        .map { case (x, y) => SpatialKey(x, y) }
+        .toList
+      val actualKeys = self.map(_._1)
+      val missingKeys = expectedKeys diff actualKeys
+
+      val missingTiles = missingKeys.map { key =>
+        (key, tile.prototype(layoutDefinition.tileLayout.tileCols, layoutDefinition.tileLayout.tileRows))
+      }
+      val allTiles = self.withContext { collection =>
+        collection.toList ++ missingTiles
```

Comment:
since you're using `++` you can concat sequences without .toList conversion `collection ++ missingTiles`

Reply:
No worries, updated.